OpenRA.Mods.Common/Widgets/Logic/ColorPickerLogic.cs

```diff
@@ -27,49 +28,53 @@ public class ColorPickerLogic : ChromeLogic
 		public ColorPickerLogic(Widget widget, ModData modData, World world, Color initialColor, string initialFaction, Action<Color> onChange,
 			Dictionary<string, MiniYaml> logicArgs)
 		{
-			if (initialFaction == null || !ChromeMetrics.TryGet("ColorPickerActorType-" + initialFaction, out string actorType))
-				actorType = ChromeMetrics.Get<string>("ColorPickerActorType");
+			var mixer = widget.Get<ColorMixerWidget>("MIXER");
+
+			// Set the initial state
+			var colorManager = world.WorldActor.Info.TraitInfo<ColorPickerManagerInfo>();
+			mixer.SetColorLimits(colorManager.HsvSaturationRange[0], colorManager.HsvSaturationRange[1], colorManager.V);
+			mixer.OnChange += () => onChange(mixer.Color);
```

Comment:
Need to remove the event handler?

Reply:
`mixer` has the same lifetime as this logic class, so shouldn't be needed.